src/libRumorSpreading/RumorStateMachine.h

```diff
@@ -0,0 +1,90 @@
+#ifndef RANDOMIZEDRUMORSPREADING_MESSAGESTATE_H
+#define RANDOMIZEDRUMORSPREADING_MESSAGESTATE_H
+
+#include "NetworkConfig.h"
+#include <array>
+#include <functional>
+#include <map>
+#include <ostream>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace RRS
+{
+
+    class RumorStateMachine
+    {
+    public:
+        // ENUMS
+        enum class State
+        {
+            UNKNOWN, // initial state where the peer 'v' doesn't know about the rumor 'r'
+            NEW, // the peer 'v' knows 'r' and counter(v,r) = m
+            KNOWN, // cooling state, stay in this state for a 'm_maxRounds' rounds
+            OLD, // final state, member stops participating in rumor spreading
+            NUM_STATES
+        };
+
+        static std::map<State, std::string> s_enumKeyToString;
+
+    private:
+        // MEMBERS
+        State m_state;
+        const NetworkConfig* m_networkConfigPtr;
+        int m_age;
+        int m_roundsInB;
+        int m_roundsInC;
+        std::unordered_map<int, int> m_memberRounds; // Member ID --> age
+
+        // METHODS
+        void advanceFromNew(const std::unordered_set<int>& membersInRound);
+
+        void advanceFromKnown();
+
+        void advanceToOld();
+
+    public:
+        // CONSTRUCTORS
+        // Default constructor. The returned state machine instance will be in an invalid state.
+        RumorStateMachine();
```

Comment:
If we shouldn't use this constructor I think you can just declare it as `RumorStateMachine() = delete;` and remove the definition in the cpp.